### PR TITLE
chore: storybook 10 upgrade

### PR DESCRIPTION
**Pull request by VeeKemp**
Ticket: https://telicent.atlassian.net/browse/TELFE-1593

## Work Done

Upgrades Storybook from 8.4.0 to 10.4.6 to unlock the modern accessibility-testing tooling (@storybook/addon-a11y + @storybook/addon-vitest), which is gated on Storybook 9+. The actual a11y addons will land in a separate follow-up PR — this one is just the platform upgrade.

The upgrade was run as the official codemod in two passes (storybook@9 upgrade then storybook@latest upgrade) because the tool refuses to skip majors. Where the codemod missed edge cases (a Jest test importing from @storybook/test, an unusual @storybook/react/* wildcard import, a tsconfig that needed updating for the new package exports field), they were fixed by hand.

## What changed

Storybook packages — bumped all @storybook/* and related deps to 10.4.6:
- Removed (merged into core): @storybook/addon-essentials, @storybook/addon-interactions, @storybook/blocks, @storybook/test, @storybook/theming, @storybook/react
- Bumped: @chromatic-com/storybook 1→5, storybook-dark-mode 4→5, @storybook/addon-designs 8→11
- Added: @storybook/addon-docs

Import rewrites across ~55 story files and .storybook/preview.tsx:
- @storybook/react → @storybook/react-vite
- @storybook/test → storybook/test
- @storybook/theming → storybook/theming
- @storybook/manager-api → storybook/manager-api

Other config
- .storybook/main.ts — addons array trimmed (removed packages now bundled into core)
- .storybook/preview.tsx — backgrounds parameter migrated from the deprecated default/values shape to the new options map (auto-applied by codemod)
- tsconfig.json — moduleResolution: "node" → "bundler" so TypeScript can resolve packages that use the modern exports field
- CopyToClipboard.test.tsx — Jest test imports waitForElementToBeRemoved from @testing-library/react instead of the now-removed @storybook/test
- FeatureMap.stories.tsx — fixed long-standing @storybook/react/* wildcard typo the codemod skipped

## What's not affected

- No public API changes to the design system itself
- No changes to consuming-app code paths (validated locally via verdaccio against  admin and search)
- Jest test suite unchanged, all existing tests still pass

## Test plan

- [x] yarn build-storybook — clean, 46s
- [x] yarn test — 248 passed, 7 skipped, 0 failed
- [x] Storybook doctor reports no issues
- [x] Built locally via verdaccio and consumed in another Telicent app — no breaking changes observed
- [x] Storybook UI in browser after merge

---

<details>
  <summary><strong>🕵️ Review Guidance</strong></summary>

---

General guidance

- Generally, approve a PR if it makes the system better, even if it's not perfect. — [Google: The Standard of Code Review](https://google.github.io/eng-practices/review/reviewer/standard.html)
- The aim of both PR AUTHOR and PR REVIEWER is to get the code merged
- Aim for consensus, defined as _everyone can live with the outcome_
- PR with changes requires 2 approvals
- PR with no changes requires 1 approvals

For PR REVIEWER:

1. Read the ticket & description
2. [Review the code](https://google.github.io/eng-practices/review/reviewer/looking-for.html)
3. Request any changes that are essential.
4. For non-essential comments:
   - Use prefixes, e.g. "**nit:** change to Pascal case"
     - **nit:** small, non-essential change
     - **obs:** just an observation, doesn't affect the PR
     - **idea:** a suggestion to think about
     - **q:** questions
   - Use modifiers, e.g. "**obs**`[pr-owner-resolve]`: Jim is also editing this file"
     - `pr-author-resolve` PR author can resolve after reading
     - `pr-author-delete` (rare) delete after reading to avoid confusion
5. try to add _at least_ a helpful comment per ~500 lines; less if the PR is already busy

For PR AUTHOR:

1. Aim for enough detail in PR description for things to go smoothly
2. After requested changes, [re-request a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#re-requesting-a-review) (so the PR shows up in [reviews-requested:@me](https://github.com/pulls?q=is%3Apr+is%3Aopen+archived%3Afalse+sort%3Aupdated-desc+review-requested%3A%40me+))

</details>

### CODE REVIEW

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -56,17 +60,17 @@ const preview: Preview = {
       },
     },
     backgrounds: {
-      default: "dark",
-      values: [
-        {
+      options: {
```
Comment (davec504) [inline .storybook/preview.tsx:63]: Why do we have two backgrounds fields, can we not just put the value  as a sibling to options?

**Review by davec504 — APPROVED** [review]

**Review by davec504 — CHANGES_REQUESTED** [review]
Lets fix the styles import

File: .storybook/preview.tsx (hunk 1 above)
Comment (VeeKemp) [reply .storybook/preview.tsx:63]: They're not duplicates — parameters.backgrounds.options declares the available choices, while initialGlobals.backgrounds.value sets the initially selected one. The selection lives in globals (not parameters) because the toolbar mutates it as users toggle, and globals persist across stories. The addon reads the current selection from globals, so moving value next to options would have no effect.

**Review by davec504 — APPROVED** [review]
